### PR TITLE
Migrate Table to RTL.

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Migrate Table to React Testing Library to prepare for React 18.

## QA

n/a

### CODE REVIEW

File: src/components/Table/Table.test.tsx [hunk 1]
```diff
@@ -1,36 +1,31 @@
...
 describe("Table", () => {
   it("renders", () => {
-    const wrapper = shallow(
+    render(
       <Table>
         <thead></thead>
       </Table>
     );
-    expect(wrapper).toMatchSnapshot();
+    expect(screen.getByRole("grid")).toMatchSnapshot();
```
Comment (bartaz) [inline src/components/Table/Table.test.tsx:13]: I guess we could use data id or container to get root element.

Looking at this, I'm not sure if `grid` is the right role for a table. It's meant for interactive grids, not static tables.

Comment (huwshimi) [reply src/components/Table/Table.test.tsx]: I've changed this to use the container.

File: src/components/Table/Table.test.tsx [hunk 2]
```diff
@@ -1,36 +1,32 @@
...
   it("renders", () => {
-    const wrapper = shallow(
+    const { container } = render(
       <Table>
         <thead></thead>
       </Table>
     );
-    expect(wrapper).toMatchSnapshot();
+    // eslint-disable-next-line testing-library/no-node-access, testing-library/no-container
+    expect(container.firstChild).toMatchSnapshot();
```
Comment (ndv99) [inline src/components/Table/Table.test.tsx:14]: This doesn't feel like a good solution to me - based on the comments from Bartek, obviously getting it by role "grid" isn't ideal either so the best thing to do here might just be to use a data id.

Comment (ndv99) [reply src/components/Table/Table.test.tsx:14]: Either that or match the container directly to the snapshot I guess

Comment (huwshimi) [reply src/components/Table/Table.test.tsx]: I've updated these tests to use a test id.

**Review by ndv99 — APPROVED** [review]
Nice one mate!

**Pull request merged**